tests/conftest.py

```diff
@@ -0,0 +1,90 @@
+def product(organization, foremanapi):
+    prod = foremanapi.create('products', {'name': str(uuid.uuid4()), 'organization_id': organization['id']})
+    yield prod
+    foremanapi.delete('products', prod)
+
+@pytest.fixture
+def yum_repository(product, organization, foremanapi):
+    repo = foremanapi.create('repositories', {'name': str(uuid.uuid4()), 'product_id': product['id'], 'content_type': 'yum', 'url': 'https://fixtures.pulpproject.org/rpm-no-comps/'})
+    yield repo
+    foremanapi.delete('repositories', repo)
+
+@pytest.fixture
+def file_repository(product, organization, foremanapi):
+    repo = foremanapi.create('repositories', {'name': str(uuid.uuid4()), 'product_id': product['id'], 'content_type': 'file', 'url': 'https://fixtures.pulpproject.org/file/'})
+    yield repo
+    foremanapi.delete('repositories', repo)
+
+@pytest.fixture
+def container_repository(product, organization, foremanapi):
+    repo = foremanapi.create('repositories', {'name': str(uuid.uuid4()), 'product_id': product['id'], 'content_type': 'docker', 'url': 'https://quay.io/', 'docker_upstream_name': 'foreman/busybox-test'})
+    yield repo
+    foremanapi.delete('repositories', repo)
+
+@pytest.fixture
+def lifecycle_environment(organization, foremanapi):
+    library = foremanapi.list('lifecycle_environments', 'name=Library', {'organization_id': organization['id']})[0]
+    lce = foremanapi.create('lifecycle_environments', {'name': str(uuid.uuid4()), 'organization_id': organization['id'], 'prior_id': library['id']})
+    yield lce
+    foremanapi.delete('lifecycle_environments', lce)
+
+@pytest.fixture
+def content_view(organization, foremanapi):
+    cv = foremanapi.create('content_views', {'name': str(uuid.uuid4()), 'organization_id': organization['id']})
+    yield cv
+    foremanapi.delete('content_views', cv)
+
+@pytest.fixture
+def activation_key(organization, foremanapi):
+    ak = foremanapi.create('activation_keys', {'name': str(uuid.uuid4()), 'organization_id': organization['id']})
+    yield ak
+    foremanapi.delete('activation_keys', ak)
+
+@pytest.fixture
+def client_environment(activation_key, content_view, lifecycle_environment, yum_repository, organization, foremanapi):
+    foremanapi.resource_action('repositories', 'sync', {'id': yum_repository['id']})
+    foremanapi.update('content_views', {'id': content_view['id'], 'repository_ids': [yum_repository['id']]})
+    foremanapi.resource_action('content_views', 'publish', {'id': content_view['id']})
+
+    library = foremanapi.list('lifecycle_environments', 'name=Library', {'organization_id': organization['id']})[0]
+    foremanapi.update('activation_keys', {'id': activation_key['id'], 'organization_id': organization['id'], 'environment_id': library['id'], 'content_view_id': content_view['id']})
+
+    yield activation_key
+
+    foremanapi.update('activation_keys', {'id': activation_key['id'], 'organization_id': organization['id'], 'environment_id': None, 'content_view_id': None})
```

Comment:
This needs https://github.com/Katello/katello/pull/11215 which is not yet in the used image